src/react/connect.ts

```diff
@@ -65,3 +67,38 @@ export const connect =
 
     return React.createElement(Component, newProps);
   };
+
+let instanceCount = 0;
+const nextInstanceName = () => `view-${instanceCount++}`;
+
+/**
+ * Higher order component for connecting a component to a stream
+ *
+ * This is the same as {@link connect }, but takes a factory for the stream. A
+ * new stream will be retrieved from the factory for every new mount of the
+ * component.
+ *
+ * @param Component Component that will receive props from the stream
+ * @param createInstanceStream Function to create the stream for each instance
+ *  of the component
+ * @returns A component that renders the original component with props retrieved
+ *  from the stream
+ * @see {@link connect}
+ */
+export const connectInstance =
```

Comment:
I am curious how to use this with a stream

Reply:
```ts
connectInstance(MyComponent, (viewInstanceId) => action$.pipe(withNamespace(viewInstanceId), ...))
```
The important part is that it accepts a function that receives a view instance name / id.